src/osp/universetypes.h

```diff
@@ -0,0 +1,44 @@
+/**
+ * Open Space Program
+ * Copyright © 2019-2020 Open Space Program Project
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#pragma once
+
+#include <Magnum/Math/Vector3.h>
+
+namespace osp::universe
+{
+
+enum class Satellite : uint32_t {};
+
+using spaceint_t = int64_t;
+
+// 1024 space units = 1 meter
+// TODO: this should vary by trajectory, but for now it's global
+const float gc_units_per_meter = 1024.0f;
```

Comment:
Were you thinking that it would be a template variable, with the template parameter being the trajectory type?

Or were you thinking that the trajectory would be a runtime lookup'ed value?

Reply:
Could you write a little bit of documentation on how the units per meter varying by trajectory would be used in practice?

Reply:
This is going to be assignable at runtime, configurable in config files. Should be useful consider for @EhWhoAmI too.

Going to link to [this old document again](https://github.com/TheOpenSpaceProgramArchive/urho-osp/wiki/Dealing-with-a-Large-Universe), but in short:
* A large galaxy can use (1 unit = 64 meters) to position stars. Who cares if stars are ± 64 meters away from each other? Milky way is about 10^21m in diameter. 64 bit int will have a max range of (2^64 * 64 = 1.18×10^21 m)
* Each solar system in that galaxy can use (1024 units = 1 meter), about 1 millimeter precision. Our solar system is about 2.9×10^11m in diameter. 64 bit ints will have a max range of (2^64 / 1024 = 1.8×10^16 m), enough for a bit of interstellar travel.

Reply:
My only concern with using 64bit ints for star positions would be how to handle interstellar travel, but I suppose one solution to that could be to have the ship travel via a series of massless stars that are positioned as a star, and are moved as the ship moves from one star to another.